server/shared/database/migrations/20200215090218-add-activity-modified-at.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+
+const filter = require('lodash/filter');
+const get = require('lodash/get');
+const head = require('lodash/head');
+const keyBy = require('lodash/keyBy');
+const map = require('lodash/map');
+const max = require('lodash/max');
+const Promise = require('bluebird');
+const transform = require('lodash/transform');
+
+const TABLE = 'activity';
+const COLUMN = 'modified_at';
+
+const isOutline = it => /\//.test(it.type);
+
+exports.up = async (qi, { DATE }) => {
+  await qi.addColumn(TABLE, COLUMN, { type: DATE });
+  await updateModifiedAt(qi);
+};
+
+exports.down = qi => qi.removeColumn(TABLE, COLUMN);
+
+async function updateModifiedAt({ sequelize }) {
+  const transaction = await sequelize.transaction();
+  const [activities, revisions] = await Promise.all([
+    getActivities(sequelize, transaction),
+    getRevisions(sequelize, transaction)
+  ]);
+  const cache = buildRevisionCache(activities, revisions);
+  const outlineActivities = filter(activities, isOutline);
+  await Promise.map(outlineActivities, activity => {
+    const modifiedAt = resolveModifiedAt(activity, cache);
+    return setModifiedAt(sequelize, transaction, activity, modifiedAt);
+  });
+  await transaction.commit();
+}
+
+async function getRevisions(sequelize, transaction) {
```

Comment:
Scope it at least per repository (Revisions can be huge for some older projects)